Neos.Flow/Tests/Functional/ObjectManagement/DependencyInjectionTest.php

```diff
@@ -333,4 +334,20 @@ public function noProxyClassIsGeneratedForPrototypeClassesWithOnlyPrototypeConst
         $object = new PrototypeClassA();
         self::assertInstanceOf(ProxyInterface::class, $object);
     }
+
+    /**
+     * @test
+     */
+    public function noProxyClassIsGeneratedForPrototypeClassesWithOptionalStraightValues(): void
+    {
+        $object = new PrototypeClassI(
+            new ValueObjectClassA('foo'),
+            null,
+            new ValueObjectClassB('bar')
+        );
+        self::assertNotInstanceOf(ProxyInterface::class, $object);
```

Comment:
No matter what you pass there – if null or a string – there wouldn't be a proxy, because that argument is passed during runtime when proxies were already built.

See my other comment for more details, I hope that helps.